wandb/sdk/wandb_lite_run.py

```diff
@@ -0,0 +1,344 @@
+                    "t": proto_util.proto_encode_to_dict(self._telemetry_obj),
+                },
+            }
+        )
+        self._job_type = job_type if not _hide_in_wb else WANDB_HIDDEN_JOB_TYPE
+        self._dir = tempfile.TemporaryDirectory()
+        if _hide_in_wb and group is None:
+            group = "weave_hidden_runs"
+        self._group = group
+        self._init_pid = os.getpid()
+        self._attach_id = self._run_name
+
+    def ensure_run(self) -> public.Run:
+        return self.run
+
+    def _add_singleton(
+        self, data_type: str, key: str, value: typing.Dict[typing.Union[int, str], str]
+    ) -> None:
+        raise NotImplementedError("Not implemented for _InMemoryLazyLiteRun")
+
+    @property
+    def dir(self) -> str:
+        return str(self._dir)
+
+    def _setup_telemetry(self) -> None:
+        self._telemetry_obj = telemetry.TelemetryRecord()
+        self._telemetry_obj_active = False
+        self._telemetry_obj_flushed = b""
+        self._telemetry_obj_dirty = False
+        with telemetry.context(self, self._telemetry_obj) as tel:
+            tel.cli_version = __version__
+            tel.python_version = platform.python_version()
+            if self._settings._jupyter:
+                tel.env.jupyter = True
+            if self._settings._ipython:
+                tel.env.ipython = True
+            if self._settings._colab:
+                tel.env.colab = True
+            if self._settings._kaggle:
+                tel.env.kaggle = True
+            if self._settings._windows:
+                tel.env.windows = True
+            tel.feature.stream_table = True
+            for module_name in telemetry.list_telemetry_imports(only_imported=True):
+                setattr(tel.imports_init, module_name, True)
+
+    def _telemetry_callback(self, telem_obj: telemetry.TelemetryRecord) -> None:
+        # TODO: support config updating, this is currently a no-op
+        pass
+
+    @property
+    def i_api(self) -> InternalApi:
+        if self._i_api is None:
+            self._i_api = InternalApi(
+                {"project": self._project_name, "entity": self._entity_name}
+            )
+        return self._i_api
+
+    @property
+    def supports_streamtable(self) -> bool:
+        # SaaS always supports streamtable
+        if self.i_api.settings("base_url").endswith("wandb.ai"):
+            return True
+        if self._server_info is None:
+            _, self._server_info = self.i_api.viewer_server_info()
+        return self._server_info.get("streamTableEnabled", False)
+
+    @property
+    def id(self) -> str:
+        return self._run_name
+
+    @property
+    def project(self) -> str:
+        return self._project_name
+
+    @property
+    def entity(self) -> str:
+        return self._entity_name
+
+    @property
+    def run(self) -> public.Run:
+        if self._run is None:
+            # TODO: decide if we want to merge an existing run
```

Comment:
can you explain this todo?

Reply:
If we're "resuming" a run, this will overwrite the `config`.  We could instead pull down the existing config and merge our current one with it.